src/parquet/column/reader.cc

```diff
@@ -165,6 +173,39 @@ bool TypedColumnReader<TYPE>::ReadNewPage() {
   return true;
 }
 
+// ----------------------------------------------------------------------
+// Batch read APIs
+
+static size_t DecodeMany(RleDecoder* decoder, int16_t* levels, size_t batch_size) {
+  size_t num_decoded = 0;
+
+  // TODO(wesm): Push this decoding down into RleDecoder itself
+  for (size_t i = 0; i < batch_size; ++i) {
+    if (!decoder->Get(levels + i)) {
```

Comment:
When does this return false? (We shouldn't fix this in this patch)

This is only for EOS errors right? Should we make the contract so the caller of the low level APIs should know and not over run it? (The lower levels can throw if the data is bad, but this is not the normal control flow path).